src/reanimated2/hook/commonTypes.ts

```diff
@@ -13,12 +14,12 @@ export interface Descriptor {
   shadowNodeWrapper: ShadowNodeWrapper;
 }
 
-export interface RefObjectFunction<T> {
+export interface RefObjectFunction<T extends Component> {
   current: T | null;
   (component?: T):
     | number // Paper
     | ShadowNodeWrapper // Fabric
     | HTMLElement; // web
 }
 
-export type AnimatedRef<T> = RefObjectFunction<T>;
+export type AnimatedRef<T extends Component> = RefObjectFunction<T>;
```

Comment:
Could we remove RefObjectFunction completely and have
```
export interface AnimatedRef<T extends Component> {
  current: T | null;
  (component?: T):
    | number // Paper
    | ShadowNodeWrapper // Fabric
    | HTMLElement; // web
}
```
instead?

Reply:
Yes we can!